### ISSUE TITLE
dead lock at Enhancer.create with cglib 3.2.5 , jdk1.8

### ISSUE THREAD

**Issue by baimashima** [cglib/cglib#120]
import java.lang.reflect.Method;
import java.util.concurrent.ExecutionException;

import net.sf.cglib.proxy.MethodInterceptor;
import net.sf.cglib.proxy.MethodProxy;

public class SimulateBlockedCglib {

	public static void main(String[] args) throws ClassNotFoundException, InterruptedException, ExecutionException {
		// change this value to see another result
		boolean isWantBlock = true;

		if (isWantBlock) {
			net.sf.cglib.proxy.Enhancer.create(EnhancerClass.class, new MethodInterceptor() {

				@Override
				public Object intercept(Object obj, Method method, Object[] args, MethodProxy proxy) throws Throwable {
					// do nothing 
					return null;
				}
			});
			System.out.println("running ");
		} else {
			net.sf.cglib.proxy.Enhancer.create(SimulateBlockedCglib.class, new MethodInterceptor() {

				@Override
				public Object intercept(Object obj, Method method, Object[] args, MethodProxy proxy) throws Throwable {
					// do nothing 
					return null;
				}
			});
			
			System.out.println("running ");
			Class.forName(EnhancerClass.class.getName()); 
		}
		Thread.sleep(100);
		System.out.println("end ");
	}

	public static class EnhancerClass {
		static {
			net.sf.cglib.proxy.Enhancer.create(EnhancerClass.class, new MethodInterceptor() {

				@Override
				public Object intercept(Object obj, Method method, Object[] args, MethodProxy proxy) throws Throwable {
					// do nothing
					return null;
				}
			});
			System.out.println("static at EnhancerClass");
		} 
	} 
}

**Comment by vlsi** [comment]
Just curious: @baimashima , is it something you have discovered in an application code?

The workaround is either avoid usage of cglib APIs in `static { ... }` blocks or use `e.setUseCache(false);` within `static { ... }`

For instance:

```java
    public static class EnhancerClass {
        static {
//            net.sf.cglib.proxy.Enhancer.create(EnhancerClass.class, new MethodInterceptor() {
//
//                @Override
//                public Object intercept(Object obj, Method method, Object[] args, MethodProxy proxy) throws Throwable {
//                    // do nothing
//                    return null;
//                }
//            });
            Enhancer e = new Enhancer();
            e.setSuperclass(EnhancerClass.class);
            e.setCallback(new MethodInterceptor() {

                @Override
                public Object intercept(Object obj, Method method, Object[] args, MethodProxy proxy) throws Throwable {
                    // do nothing
                    return null;
                }
            });
            e.setUseCache(false);
            e.create();
            System.out.println("static at EnhancerClass");
        }
```

I think cglib can be patched to automatically avoid such a case, however it does look like an edge case.

**Comment by baimashima** [comment]
Yes, I found this issue because a rookie mistake usage , but this code work well with cglib 3.1. So I consider that is an issue with 3.2.5 version.
